sdk/servicebus/azure-servicebus/azure/servicebus/_transport/_pyamqp_transport.py

```diff
@@ -804,15 +807,26 @@ def settle_message_via_receiver_link(
             if settle_operation == MESSAGE_DEFER:
                 return handler.settle_messages(
                     message._delivery_id,
+                    message._delivery_tag,
                     'modified',
                     delivery_failed=True,
                     undeliverable_here=True
                 )
         except AttributeError as ae:
             raise RuntimeError("handler is not initialized and cannot complete the message") from ae
 
+        except AMQPLinkError as le:
+            if (
+                le.condition == ErrorCondition.InternalError
+                and isinstance(le.description, str)
+                and le.description.startswith("Delivery tag")
+            ):
+                raise RuntimeError("Link error occurred during settle operation.") from le
```

Comment:
curious about why we're doing RuntimeError here and SBConnectionError down below?

Reply:
Issue is that uAMQP raises a `RuntimeError` for this and is caught [here](https://github.com/Azure/azure-sdk-for-python/blob/2beea7089d309b16e2080fe22feb807d65ee4d23/sdk/servicebus/azure-servicebus/azure/servicebus/_servicebus_receiver.py#L542) when the code moves on to settle over the mgmt link. If I raise a `ServiceBusConnectionError` other things that are not related to settle would also land up there